packages/react-components/src/components/chart/utils/getInfo.ts

```diff
@@ -33,36 +46,126 @@ export const calculateTimeStamp = (xInPixel: number, widthInPixel: number, viewp
+  count: number,
+  graphic: InternalGraphicComponentGroupOption[],
+  setGraphic: Dispatch<SetStateAction<InternalGraphicComponentGroupOption[]>>,
+  viewport?: Viewport,
+  chart?: EChartsType
+) => {
+  const uId = uuid();
+  const timestampInMs = calculateTimeStamp(e.offsetX, size?.width, viewport);
+  const boundedX = setXWithBounds(size, e.offsetX ?? 0);
+  const newTC = {
+    id: `trendCursor-${uId}`,
+    $action: 'merge',
+    type: 'group' as const,
+    timestampInMs,
+    children: [
+      {
+        type: 'line',
+        z: trendCursorZIndex,
+        id: `line-${uId}`,
+        draggable: 'horizontal' as const,
+        shape: {
+          x1: boundedX,
+          x2: boundedX,
+          y1: DEFAULT_MARGIN,
+          y2: (size?.height ?? 0) - DEFAULT_MARGIN,
+        },
+        style: {
+          stroke: trendCursorLineColor,
+          lineWidth: trendCursorLineWidth,
+        },
+        ondrag: (event: ChartEventType) => {
+          const graphicIndex = graphic.findIndex((g) => g.children[0].id === event.target.id);
+          const timeInMs = calculateTimeStamp(event.offsetX ?? 0, size?.width, viewport);
+          graphic[graphicIndex].children[1].x = setXWithBounds(size, event.offsetX ?? 0);
+          graphic[graphicIndex].children[2].x = setXWithBounds(size, event.offsetX ?? 0) + 40;
+
+          graphic[graphicIndex].children[1].style = {
+            ...graphic[graphicIndex].children[1].style,
+            text: getTrendCursorHeaderTimestampText(
+              timeInMs,
+              (graphic[graphicIndex].children[1] as GraphicComponentTextOption).style?.text
+            ),
+          };
+          graphic[graphicIndex].timestampInMs = calculateTimeStamp(e.offsetX ?? 0, size?.width, viewport);
+          chart?.setOption({ graphic });
+          setGraphic(graphic);
+        },
+      },
+      {
+        type: 'text',
+        z: trendCursorZIndex + 1,
+        id: `text-${uId}`,
+        x: boundedX,
+        style: {
+          y: DEFAULT_MARGIN,
+          text: getTrendCursorHeaderTimestampText(timestampInMs, `{title|Trend cursor ${count + 1}  }`),
+          lineHeight: 16,
+          fill: trendCursorHeaderTextColor,
+          align: 'center',
+          rich: {
+            title: {
+              width: trendCursorHeaderWidth,
+              backgroundColor: trendCursorHeaderColors[count],
+              height: 20,
+              fontSize: 12,
+            },
+            timestamp: {
+              width: trendCursorHeaderWidth,
+              backgroundColor: trendCursorHeaderBackgroundColor,
+              height: 15,
+              fontSize: 9,
+              FontWeight: 'bold',
+            },
+          },
+        },
       },
-      style: {
-        stroke: trendCursorLineColor,
-        lineWidth: trendCursorLineWidth,
+      {
+        id: `image-${uId}`,
+        type: 'image',
+        z: trendCursorZIndex + 1,
+        x: boundedX + 40,
+        y: DEFAULT_MARGIN + 2.5,
```

Comment:
What do these numbers mean? Can we make their meaning clear?

Reply:
these are positional offsets,  will create a separate constants